src/DepositHandler.sol

```diff
@@ -21,9 +23,12 @@ contract DepositHandler {
         noReenter
         returns (bool success, bytes memory returnData)
     {
-        (success, returnData) = callee.call(data);
+        if (callee.code.length != 0) {
+            (success, returnData) = callee.call(data);
+        }
     }
 
+    // NOTE: The gateway should always destroy the `DepositHandler` in the same runtime context that deploys it.
     function destroy(address etherDestination) external noReenter {
         selfdestruct(payable(etherDestination));
```

Comment:
```suggestion
    function destroy(address etherDestination) external {
        if (_lockedStatus == IS_LOCKED) revert IsLocked();

        selfdestruct(payable(etherDestination));
```
Instead of running the modifier? Or does selfdestruct cause the `_lockedStatus` to be set back to `0` and give us a refund?

Reply:
I need to test once the hardhat PR is merged. Then we'll revert or correct.